File: fix_links.js
const fs = require('fs');
const path = require('path');

const files = ["index.html", "Food.html", "Travel.html", "Entertainment.html", "Health.html", "Voucher.html"];

const mapping = [
    { text: "Hot Deals", link: "index.html" },
    { text: "Trang chủ", link: "index.html" },
    { text: "Du lịch", link: "Travel.html" },
    { text: "Ẩm thực", link: "Food.html" },
    { text: "Giải trí", link: "Entertainment.html" },
    { text: "Sức khỏe", link: "Health.html" },
    { text: "Voucher", link: "Voucher.html" }
];

files.forEach(filename => {
    const filePath = path.join(__dirname, filename);
    if (!fs.existsSync(filePath)) return;

    let content = fs.readFileSync(filePath, 'utf-8');

    mapping.forEach(({ text, link }) => {
        // Regex to find: href="..." followed by optional classes/spaces then >text</a>
        const regex = new RegExp(`href="[^"]*"([^>]*>${text}</a>)`, 'g');
        content = content.replace(regex, `href="${link}"$1`);
    });

    fs.writeFileSync(filePath, content, 'utf-8');
});

console.log("Links updated successfully.");

File: fix_links.py
import os
import re

files = ["index.html", "Food.html", "Travel.html", "Entertainment.html", "Health.html", "Voucher.html"]
mapping = {
    "Hot Deals": "index.html",
    "Trang chủ": "index.html",
    "Du lịch": "Travel.html",
    "Ẩm thực": "Food.html",
    "Giải trí": "Entertainment.html",
    "Sức khỏe": "Health.html",
    "Voucher": "Voucher.html"
}

for filename in files:
    if not os.path.exists(filename):
        continue
    with open(filename, 'r', encoding='utf-8') as f:
        content = f.read()
    
    for text, link in mapping.items():
        # Match <a ...>text</a> and replace href="#" with link
        pattern = rf'href="#"([^>]*>{text}</a>)'
        content = re.sub(pattern, f'href="{link}"\\1', content)
        
    with open(filename, 'w', encoding='utf-8') as f:
        f.write(content)

print("Links updated successfully.")
